File: mkfuncs.pl
#! /usr/bin/perl
use strict;

my $state = 0;
my $def;
my $params;

while (<>) {
	if (/^\tpublic\s+(.*)/) {
		$def = "public $1";
		$state = 1;
		$params = 0;
	} elsif ($state == 1 and /(\w+)\s*\(/) {
		$def .= " $1 LESSPARAMS((";
		$state = 2;
	} elsif ($state == 2) {
		if (/^{/) {
			$def .= 'VOID_PARAM' if not $params;
			print "$def));\n";
			$state = 0;
		} elsif (/^\s*([^;]*)/) {
			$def .= ', ' if substr($def,-1) ne '(';
			$def .= $1;
			$params = 1;
		}
	}
}

File: mkfuncs.py
#!/usr/bin/env python

import fileinput
import re

definition = ''
state = 0
params = 0

for line in fileinput.input():
        if test := re.search(r'^\tpublic\s+(.*)', line):
            definition = 'public ' + test.group(1)
            state = 1
            params = 0
        elif (state == 1) and (test := re.search(r'(\w+)\s*\(', line)):
            definition = '{0} LESSPARAMS (('.format(test.group(1))
            state = 2
        elif state == 2:
            if re.search(r'^{', line):
                if not params: definition += 'VOID_PARAM'
                print(f'{definition}));')
                state = 0
            elif test := re.search(r'^\s*([^;]*)', line):
                if (definition[-1:] != '('): definition += ', ' 
                definition += test.group(1) 
                params = 1
